service/controller/appcatalog/resource/appcatalogentry/resource.go

```diff
@@ -116,3 +116,37 @@ func (r *Resource) getIndex(ctx context.Context, storageURL string) (index, erro
 
 	return i, nil
 }
+
+func (r *Resource) getRestrictions(ctx context.Context, storageURL, name, version string) (*v1alpha1.AppCatalogEntrySpecRestrictions, error) {
+	mainURL := fmt.Sprintf("%s/%s-%s.tgz-meta/main.yaml", strings.TrimRight(storageURL, "/"), name, version)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting main.yaml from %#q", mainURL))
+
+	// We use https in catalog URLs so we can disable the linter in this case.
+	resp, err := http.Get(mainURL) // #nosec
```

Comment:
I don't want to prematurely optimize this but we also need to watch how many HTTP requests we make per reconciliation loop.

We may need some kind of HTTP cache. Especially for China where calls to GitHub are slow.

Reply:
Having metrics around the HTTP call is nice, but applying cache is a quite big change. 
I would like to focus on the idea of the metric first, and we can think about applying cache after this PR. 

First, I will look into other operators on how to apply HTTP metrics inside the controller itself.

Reply:
Agree the cache is a bigger change. For now we only generate CRs for the giantswarm and playground catalogs but we may need more later.

Good to add metrics. Maybe we can do similar to helmclient for pulling tarballs?

Reply:
I added histogram type as metrics, just like pulling tarball from helmclient. 
```
# TYPE app_operator_appcatalogentry_event histogram
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.005"} 171
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.01"} 173
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.025"} 174
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.05"} 174
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.1"} 347
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.25"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="0.5"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="1"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="2.5"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="5"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="10"} 348
app_operator_appcatalogentry_event_bucket{event="pull_metadata_file",le="+Inf"} 348
```
WDYT?

Reply:
Nice! LGTM